roles/rpm_ostree_status_verify/tasks/main.yml

```diff
@@ -0,0 +1,58 @@
+---
+# vim: set ft=ansible:
+#
+#  This role validates rpm-ostree status output.
+#
+#  Parameters:
+#   deployment (int) - index of deployment to verify
+#   expected (dict) - a dict of expected key/value pairs
+#
+- name: Fail if deployment or expected is undefined
+  fail:
+    msg: "deployment or expected is undefined"
+  when: expected is undefined or
+        deployment is undefined
+
+- name: Get rpm-ostree status output
+  command: rpm-ostree status --json
+  register: ros_status
+
+- name: Set json output
+  set_fact:
+     ros_json: "{{ ros_status.stdout | from_json }}"
+
+- name: Set ros variable if deployment 0 is booted
+  set_fact:
+    ros_booted: "{{ ros_json['deployments'][0] }}"
+    ros_not_booted: "{{ ros_json['deployments'][1] if ros_json['deployments'][1] is defined else false }}"
+  when: ros_json['deployments'][0] is defined and ros_json['deployments'][0]['booted']
+
+- name: Set ros variable if deployment 1 is booted
+  set_fact:
+    ros_booted: "{{ ros_json['deployments'][1] }}"
+    ros_not_booted: "{{ ros_json['deployments'][0] if ros_json['deployments'][0] is defined else false }}"
+  when: ros_json['deployments'][1] is defined and ros_json['deployments'][1]['booted']
+
+- name: Set number of deployments
+  set_fact:
+     ros_num_deployments: "{{ ros_json['deployments'] | length }}"
+
+- name: Verify number of deployments
+  fail:
+    msg: |
+      "Number of deployments is incorrect.
+       Expected: {{ num_deployments }}.
+       Actual {{ ros_num_deployments }}"
+  when: num_deployments is defined and {{ num_deployments | int }} != {{ ros_num_deployments | int }}
+
+- name: Fail if deployment does not exist
+  fail:
+    msg: "deployment {{ deployment }} does not exist"
+  when: ros_json['deployments'][deployment] is undefined
+
+- name:  Fail if deployment {{ deployment }} values are incorrect
```

Comment:
I was a bit confused why you used a `dict` here, but seeing it used in practice in the main test file made me understand.

Great idea and very flexible!